### DISCUSSION TITLE
Regarding openCV installation dependencies

### DISCUSSION THREAD

**Discussion by zirid** [microsoft/vcpkg#35145] [General]
I installed opencv for windows with cuda using the following cmd :

` vcpkg install opencv[cuda,cudnn,dnn,dnn-cuda,ipp,opengl,world]`

I am curious whether it is necessary to add additional dll's  when opencv installation already generated ` opencv_world dll   ` ?

My test app can't run without adding all the generated dll's . Is this normal?

![opencv_vcpkg](https://github.com/microsoft/vcpkg/assets/31682698/5f710811-1920-4c6b-8329-f43d720fe296)

**Comment by vicroms** [comment]
Short answer: Yes, you need all those DLLs. 

You have built OpenCV and its dependencies as [dynamic link libraries](https://learn.microsoft.com/windows/win32/dlls/dynamic-link-libraries), each DLL contains the code for its respective library's functions/classes.

If you'd rather have a single executable, you may look into building your dependencies as static libraries instead.
